# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/test/unit/head_cookies.js
# Commit: d01a1a055fec
# Full Hash: d01a1a055fece12b906b938954f984bd69278ef7
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-30 20:49:53
# Regressor Bug: 1960824
# File Overlap Count: 1
# Description:
#   Bug 1960824 - Introduce nsICookie::SAMESITE_UNSET, r=edgul,webdriver-reviewers,extension-reviewers,cookie-reviewers,valentin,Sasha,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246638
# ==============================================================================

diff -r 307ff391dd02 -r d01a1a055fec netwerk/test/unit/head_cookies.js
--- a/netwerk/test/unit/head_cookies.js	Wed Apr 30 08:53:12 2025 +0000
+++ b/netwerk/test/unit/head_cookies.js	Wed Apr 30 08:59:18 2025 +0000
@@ -204,8 +204,7 @@
   isHttpOnly,
   inBrowserElement = false,
   originAttributes = {},
-  sameSite = Ci.nsICookie.SAMESITE_NONE,
-  rawSameSite = Ci.nsICookie.SAMESITE_NONE,
+  sameSite = Ci.nsICookie.SAMESITE_UNSET,
   schemeMap = Ci.nsICookie.SCHEME_UNSET,
   isPartitioned = false
 ) {
@@ -222,10 +221,12 @@
   this.inBrowserElement = inBrowserElement;
   this.originAttributes = originAttributes;
   this.sameSite = sameSite;
-  this.rawSameSite = rawSameSite;
   this.schemeMap = schemeMap;
   this.isPartitioned = isPartitioned;
 
+  // For pre version 15 compatibility:
+  this.rawSameSite = sameSite;
+
   let strippedHost = host.charAt(0) == "." ? host.slice(1) : host;
 
   try {
@@ -767,6 +768,79 @@
 
       break;
     }
+    case 15: {
+      if (!exists) {
+        this.db.executeSimpleSQL(
+          "CREATE TABLE moz_cookies (                     \
+            id INTEGER PRIMARY KEY,                       \
+            originAttributes TEXT NOT NULL DEFAULT '',    \
+            name TEXT,                                    \
+            value TEXT,                                   \
+            host TEXT,                                    \
+            path TEXT,                                    \
+            expiry INTEGER,                               \
+            lastAccessed INTEGER,                         \
+            creationTime INTEGER,                         \
+            isSecure INTEGER,                             \
+            isHttpOnly INTEGER,                           \
+            inBrowserElement INTEGER DEFAULT 0,           \
+            sameSite INTEGER DEFAULT 0,                   \
+            schemeMap INTEGER DEFAULT 0,                  \
+            isPartitionedAttributeSet INTEGER DEFAULT 0,  \
+            CONSTRAINT moz_uniqueid UNIQUE (name, host, path, originAttributes))"
+        );
+
+        this.db.executeSimpleSQL("PRAGMA journal_mode = WAL");
+        this.db.executeSimpleSQL("PRAGMA wal_autocheckpoint = 16");
+      }
+
+      this.stmtInsert = this.db.createStatement(
+        "INSERT INTO moz_cookies (        \
+           name,                          \
+           value,                         \
+           host,                          \
+           path,                          \
+           expiry,                        \
+           lastAccessed,                  \
+           creationTime,                  \
+           isSecure,                      \
+           isHttpOnly,                    \
+           inBrowserElement,              \
+           originAttributes,              \
+           sameSite,                      \
+           schemeMap,                     \
+           isPartitionedAttributeSet      \
+         ) VALUES (                       \
+           :name,                         \
+           :value,                        \
+           :host,                         \
+           :path,                         \
+           :expiry,                       \
+           :lastAccessed,                 \
+           :creationTime,                 \
+           :isSecure,                     \
+           :isHttpOnly,                   \
+           :inBrowserElement,             \
+           :originAttributes,             \
+           :sameSite,                     \
+           :schemeMap,                    \
+           :isPartitionedAttributeSet)"
+      );
+
+      this.stmtDelete = this.db.createStatement(
+        "DELETE FROM moz_cookies          \
+           WHERE name = :name AND host = :host AND path = :path AND \
+                 originAttributes = :originAttributes"
+      );
+
+      this.stmtUpdate = this.db.createStatement(
+        "UPDATE moz_cookies SET lastAccessed = :lastAccessed \
+           WHERE name = :name AND host = :host AND path = :path AND \
+                 originAttributes = :originAttributes"
+      );
+
+      break;
+    }
 
     default:
       do_throw("unrecognized schemaVersion!");
@@ -913,6 +987,29 @@
         );
         break;
 
+      case 15:
+        this.stmtInsert.bindByName("name", cookie.name);
+        this.stmtInsert.bindByName("value", cookie.value);
+        this.stmtInsert.bindByName("host", cookie.host);
+        this.stmtInsert.bindByName("path", cookie.path);
+        this.stmtInsert.bindByName("expiry", cookie.expiry);
+        this.stmtInsert.bindByName("lastAccessed", cookie.lastAccessed);
+        this.stmtInsert.bindByName("creationTime", cookie.creationTime);
+        this.stmtInsert.bindByName("isSecure", cookie.isSecure);
+        this.stmtInsert.bindByName("isHttpOnly", cookie.isHttpOnly);
+        this.stmtInsert.bindByName("inBrowserElement", cookie.inBrowserElement);
+        this.stmtInsert.bindByName(
+          "originAttributes",
+          ChromeUtils.originAttributesToSuffix(cookie.originAttributes)
+        );
+        this.stmtInsert.bindByName("sameSite", cookie.sameSite);
+        this.stmtInsert.bindByName("schemeMap", cookie.schemeMap);
+        this.stmtInsert.bindByName(
+          "isPartitionedAttributeSet",
+          cookie.isPartitioned
+        );
+        break;
+
       default:
         do_throw("unrecognized schemaVersion!");
     }